### PR TITLE
[WIP] feat: add ORA filter for student_view rendering intervention

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
**TL;DR -** Add an Open edX Filter for `student_view`

This PR adds a new extension point for the `student_view` handler. By triggering an [Open edX Filter](https://github.com/openedx/openedx-filters) before rendering the block's `student_view` we could:
- Modify the template/context being rendered by the student view during runtime -- eg. by using a plugin that extends ORA
- Access the block status for the student

What motivates us to add this filter is this case specifically:
_As a professor or student, I want to check my essay with an external service like Turnitin for plagiarism._

More on the filter's definition in this PR: https://github.com/openedx/openedx-filters/pull/107

**What changed?**

- A new Open edX Filter is triggered before returning the student fragment. When there is no filter configured, this change acts as a noop.
- We added the block object to the context so the filter can access the needed data, which can be a lot when rendering a template. We're open to suggestions on how we can improve it!
- Added `openedx-filters` to the requirements list

**Developer Checklist**

- [ ] Reviewed the [release process](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)
- [ ] Translations and JS/SASS compiled
- [ ] Bumped version number in [openassessment/\_\_init\_\_.py](https://github.com/openedx/edx-ora2/blob/master/openassessment/__init__.py#L4) and [package.json](https://github.com/openedx/edx-ora2/blob/master/package.json#L3)

**Testing Instructions**

This is a proof of concept, so testing it can be wacky. Here's a list of steps you can follow so you can test the ongoing implementation:
1. Install the needed libraries:
```
pip install git+https://github.com/openedx/openedx-filters.git@MJG/ora-student-view-filter
pip install git+https://github.com/eduNEXT/edx-ora2.git@MJG/ora-student-view-filter
pip install git+https://github.com/eduNEXT/platform-plugin-turnitin.git@MJG/turnitin-pipeline-step
```
2. Add the following configuration:
```
    OPEN_EDX_FILTERS_CONFIG = {
      "org.openedx.learning.ora.student_view.render.started.v1": {
          "fail_silently": False,
          "pipeline": [
              "platform_plugin_turnitin.extensions.filters.CheckTurnitinForPlagiarism",
          ]
      },
    }
```
3. Restart services
4. Configure an ORA block for text submissions
5. Submit your assessment

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @mariajgrimaldi!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Comment by itsjeyd** [comment]
Thanks for the contribution @mariajgrimaldi! :)

**Comment by itsjeyd** [comment]
Hey @mariajgrimaldi, just checking in to see if you're still planning to continue working on this PR?

**Comment by mariajgrimaldi** [comment]
@itsjeyd: thanks for reaching out! I'll close this for the time being and re-open it when needed.

**Comment by openedx-webhooks** [comment]
@mariajgrimaldi Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-ora2&entry.867055334=https://github.com/openedx/edx-ora2/pull/2000&entry.1484655318=https://github.com/mariajgrimaldi&entry.752974735=2023-07-27+23:58&entry.1917517419=2023-09-13+13:36&entry.2133058324=No) so we can improve your experience in the future.

**Comment by openedx-webhooks** [comment]
@mariajgrimaldi Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-ora2&entry.867055334=https://github.com/openedx/edx-ora2/pull/2000&entry.1484655318=https://github.com/mariajgrimaldi&entry.752974735=2023-07-27+23:58&entry.1917517419=2023-09-13+13:36&entry.2133058324=No) so we can improve your experience in the future.

**Comment by itsjeyd** [comment]
@mariajgrimaldi Sounds good, thanks 🙂
